src/pycram/yarp_utils/yarp_joint_state_updater.py

```diff
@@ -0,0 +1,324 @@
+        self.robot_name_yarp = None
+
+        self.state_torso_port_name = None
+        self.state_right_arm_port_name = None
+        self.state_left_arm_port_name = None
+        self.state_head_port_name = None
+
+        self.torso_joints_name = None
+        self.right_arm_joints_name = None
+        self.left_arm_joints_name = None
+        self.head_joints_name = None
+
+        self.state_torso_port = yarp.BufferedPortBottle()
+        self.state_right_arm_port = yarp.BufferedPortBottle()
+        self.state_left_arm_port = yarp.BufferedPortBottle()
+        self.state_head_port = yarp.BufferedPortBottle()
+
+    def configure(self, rf):
+        self.module_name = rf.check("name",
+                                    yarp.Value("pycramICubStatePublisher"),
+                                    "module name (string)").asString()
+
+        self.robot_name_yarp = rf.check("robot",
+                                    yarp.Value("icub"),
+                                    "module name (string)").asString()
+
+        self.state_torso_port_name = self.getName("/state/torso:i")
+        self.state_right_arm_port_name = self.getName("/state/right_arm:i")
+        self.state_left_arm_port_name = self.getName("/state/left_arm:i")
+        self.state_head_port_name = self.getName("/state/head:i")
+
+
+        # Create handle port to read message
+        if not self.handle_port.open('/' + self.module_name):
+            print("Can't open the port correctly")
+            return False
+
+        if not self.state_torso_port.open(self.state_torso_port_name):
+            print(f'Can\'t open the port correctly {self.state_torso_port_name}')
+            return False
+
+        if not self.state_right_arm_port.open(self.state_right_arm_port_name):
+            print(f'Can\'t open the port correctly {self.state_right_arm_port_name}')
+            return False
+
+        if not self.state_left_arm_port.open(self.state_left_arm_port_name):
+            print(f'Can\'t open the port correctly {self.state_left_arm_port_name}')
+            return False
+
+        if not self.state_head_port.open(self.state_head_port_name):
+            print(f'Can\'t open the port correctly {self.state_head_port_name}')
+            return False
+        if not self.connect_ports():
+            print("Error in connecting ports. Make sure that the robot is already up")
+            return False
+
+
+        self.torso_joints_names =  RobotDescription.current_robot_description.get_actuated_joint_names("torso")
+        self.right_arm_joints_names = RobotDescription.current_robot_description.get_actuated_joint_names("right_arm")
+        self.left_arm_joints_names = RobotDescription.current_robot_description.get_actuated_joint_names("left_arm")
+        self.head_joints_names = ["neck_pitch","neck_roll","neck_yaw","eyes_tilt","r_eye_pan_joint","l_eye_pan_joint"]
+
+        if self.torso_joints_names is None:
+            print("Error in joint names")
```

Comment:
That is a bit misleading, just say there are no joint names defined in the description

Reply:
I had it for some tests, but no need for it. you are right